File: 3451-string-compression-iii/3451-string-compression-iii.js
/**
 * @param {string} word
 * @return {string}
 */
var compressedString = function (word) {
    word = word.split("")
    let str = ''
    let i = 0
    while (i < word.length) {
        let j = i + 1
        while (j < word.length && j - i < 9 && word[i] == word[j]) {
            j = j + 1
        }
        str = str + `${j - i}${word[i]}`
        i = j
    }
    return str
};

File: 3451-string-compression-iii/3451-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        word = list(word)
        i = 0
        s = ''
        while i < len(word):
            j = i + 1
            while j < len(word) and j - i < 9 and word[i] == word[j]:
                j = j + 1
            s = s + f"{j-i}" + word[i]
            i = j
        return s